yul-format/main.cpp

```diff
@@ -0,0 +1,139 @@
+#include <libyul/AsmParser.h>
+#include <libyul/AsmPrinter.h>
+#include <libyul/AssemblyStack.h>
+#include <libyul/Dialect.h>
+#include <libyul/backends/evm/EVMDialect.h>
+
+#include <liblangutil/ErrorReporter.h>
+#include <liblangutil/SourceReferenceFormatterHuman.h>
+
+#include <libdevcore/CommonIO.h>
+
+#include <cstdlib>
+#include <boost/optional.hpp>
+#include <boost/program_options.hpp>
+
+using boost::optional;
+using namespace std;
+
+namespace po = boost::program_options;
+
+boost::optional<std::string> prettyPrint(
+	std::string const& _sourceCode,
+	std::string const& _sourceName,
+	yul::EVMDialect const& _evmDialect,
+	langutil::ErrorReporter& _errorReporter
+)
+{
+	langutil::EVMVersion const evmVersion = *langutil::EVMVersion::fromString("petersburg");
+	yul::EVMDialect const& evmDialect = yul::EVMDialect::strictAssemblyForEVM(evmVersion);
+	yul::Parser parser{_errorReporter, evmDialect};
+	langutil::CharStream source{_sourceCode, _sourceName};
+	auto scanner = make_shared<langutil::Scanner>(source);
```

Comment:
Did you try using the AssemblyStack?

Reply:
I  checked AssemblyStack, and AFAIR, it was also doing analysis, a step that we actually are not interested in, are we?

Also, with your comments from todays meeting (Having `AssemblyStack` not just parse & analyze, but also generate code - which totally makes sense to me, that's why it's called "stack"), I think we shouldn't use AssemblyStack unless we provide it with a public member function to just parse (no sema/...) and also provide access to the internally generated AST. What do you think?

Reply:
If you can use AssemblyStack here, I think you should. Also doing analysis does not hurt, does it? The earlier analysis errors are caught the better. We could also introduce a `parse` method into the AssemblyStack that just does the first step.

Reply:
I agree, but in this case, we only want to parse and get a pretty-printed AST regardless of correctness. Otherwise I'd definitely agree. Will be adding an AssemblyStack::parse() function. :)